tests/client-sdk/agents/test_agents.py

```diff
@@ -177,7 +179,8 @@ def test_builtin_tool_brave_search(llama_stack_client, agent_config):
     assert "tool_execution>" in logs_str
     assert "Tool:brave_search Response:" in logs_str
     assert "obama" in logs_str.lower()
-    assert "No Violation" in logs_str
+    if len(agent_config["input_shields"]) > 0:
+        assert "No Violation" in logs_str
```

Comment:
we should probably also just say "No shield present" instead of no message. That way, we can keep reinforcing that we expect shields but none were provided.

Reply:
We could follow up on this in our agent loop refactor (e.g. decide whether we should always execute a shield_call step even if no shields are present).